### PR TITLE
Defaults for ImageClassification API

### PR DESCRIPTION
**Pull request by harshithapv**
Changed EarlyStopping to run by default with ExponentialLR Decay for learning rate scheduling.
This combination seems to give most optimal results with a trade-off balance between the accuracy and training time.

### CODE REVIEW

File: src/Microsoft.ML.Vision/ImageClassificationTrainer.cs [hunk 1]
```diff
@@ -389,7 +389,7 @@ public sealed class Options : TrainerInputBaseWithLabel
             /// Callback to report statistics on accuracy/cross entropy during training phase.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Callback to report metrics during training and validation phase.", SortOrder = 15)]
-            public Action<ImageClassificationMetrics> MetricsCallback = null;
+            public Action<ImageClassificationMetrics> MetricsCallback = (metrics) => Console.WriteLine(metrics);
```
Comment (eerhardt) [inline src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: The ML.NET library shouldn't be writing to the console. We are going to be used in a lot more places than just a console app.

Comment (codemzs) [reply src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: how about to stdout all the time?

Comment (eerhardt) [reply src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: I don't view this as any different than the `MLContext.Log` property. We don't dump information to the console in it by default. I don't think we should be here either.

Comment (harshithapv) [reply src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: Addressed this in my recent commit.

File: src/Microsoft.ML.Vision/ImageClassificationTrainer.cs [hunk 2]
```diff
@@ -419,7 +419,7 @@ public sealed class Options : TrainerInputBaseWithLabel
             /// Validation set.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Validation set.", SortOrder = 15)]
-            public IDataView ValidationSet;
+            public IDataView ValidationSet = null;
```
Comment (eerhardt) [inline src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: This is unnecessary. Reference fields are set to `null` by default in .NET.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/ImageClassification/ImageClassificationDefault.cs [hunk 3]
```diff
@@ -29,13 +29,15 @@ public static void Example()
             //Download the image set and unzip
             string finalImagesFolderName = DownloadImageSet(
                 imagesDownloadFolderPath);
+            //string finalImagesFolderName = "flower_photos";
```
Comment (codemzs) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/ImageClassification/ImageClassificationDefault.cs]: >//string finalImagesFolderName = "flower_photos"; [](start = 12, length = 49)

remove

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/ImageClassification/ImageClassificationDefault.cs [hunk 4]
```diff
@@ -60,7 +62,7 @@ public static void Example()
                 IDataView testDataset = trainTestData.TestSet;
 
                 var pipeline = mlContext.MulticlassClassification.Trainers
-                        .ImageClassification(featureColumnName:"Image", validationSet:testDataset)
+                        .ImageClassification(featureColumnName:"Image", validationSet:null)
```
Comment (codemzs) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/MulticlassClassification/ImageClassification/ImageClassificationDefault.cs]: >, validationSet:null [](start = 70, length = 20)

no need, default is null.

File: src/Microsoft.ML.Vision/ImageClassificationTrainer.cs [hunk 5]
```diff
@@ -532,6 +532,12 @@ internal ImageClassificationTrainer(IHostEnvironment env, Options options)
                 options.ValidationSetBottleneckCachedValuesFileName = _options.ValidationSetBottleneckCachedValuesFileName;
             }
 
+            if ( options.MetricsCallback == null )
```
Comment (codemzs) [inline src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: >  [](start = 16, length = 1)

space

Comment (codemzs) [inline src/Microsoft.ML.Vision/ImageClassificationTrainer.cs]: >  [](start = 48, length = 1)

space
